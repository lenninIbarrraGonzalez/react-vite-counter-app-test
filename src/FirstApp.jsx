import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name, person , taxes}) => {
  // console.log(props);
  // const { first_name, last_name, cc, age} = person;
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <code>{ JSON.stringify( newMessage ) }</code> */}
      <p>{ subTitle }</p>
      <p>{ name }</p>
      {/* <p>{ JSON.stringify(persona) }</p> */}
      {/* <p>{ first_name }</p>
      <p>{ last_name }</p> */}
      {/* <p>{ taxes }</p> */}
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  taxes: PropTypes.number
}

FirstApp.defaultProps = {
  name: 'Fernando Herrera',
  subTitle: 'No hay subtítulo',
  title: 'No hay título',
  taxes: 0
}
