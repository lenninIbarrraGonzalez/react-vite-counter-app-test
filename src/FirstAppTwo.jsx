const newMessage = 'Este es mi titulo',
newNumber = 123,
newArray = [1,2,3,4,5,6,7,8,9],
newBoolean = true,
newObject = {
    first_name:'Lennin',
    last_name: 'Ibarra'
}

const getName = ( name ) => name;

export const FirstAppTwo = ({title, subTitle}) => {
    return (
        <>
            <h1>{ title }</h1>
            <h2>{subTitle}</h2>
            <h3>{newMessage}</h3>
            <h3>{ newNumber }</h3>
            <h3>{ newArray }</h3>
            <h3>{ newBoolean }</h3>
            {/* <h3>{ newObject }</h3> genera un error no se puede renderizar un objeto directamente */}
            <code>{ JSON.stringify(newObject) }</code>
            <h3>{ getName('Valentina') }</h3>
        </>
        
    )
}

