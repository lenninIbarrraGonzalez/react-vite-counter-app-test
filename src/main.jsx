import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
// import { FirstAppTwo } from './FirstAppTwo';
// import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';
import './styles.css';


const person = {
    first_name:'Lenin',
    last_name:'Ibarra',
    cc:'1307112002',
    age: 23
}

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={ 20 } />
        {/* <FirstApp title='Este es mi titulo' person={person} taxes={52000}/> */}
        {/* <FirstAppTwo title= "Este es mi titulo" subTitle="Subtitlo enviado desde las props"/> */}
    </React.StrictMode>
);


