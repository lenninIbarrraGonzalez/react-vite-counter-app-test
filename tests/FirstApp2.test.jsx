import { getByTestId, render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

//Esta es la fomra como se deben hacer la pruebas
describe('Pruebas en <FirstApp2 />', () => {
    const title    = 'Este es mi titulo';
    const subTitle = 'Subtitlo enviado desde las props'

    test('debe hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "Este es mi titulo"', () => {
        render(<FirstApp title={title}/>);
        // para ver el objeto
        // screen.debug();
        expect(screen.getAllByText(title)).toBeTruthy();
        // para verificar que no contenga el texto
        // expect(screen.getAllByText(title)).not.toBeTruthy();
    });

    test('debe de mostrar el titulo en un h1', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(title);
    });

    test('debe mostrar el subtitulo enviado por props', () => {
        render(<FirstApp
            title={title}
            subTitle={subTitle}
        />);
        expect(screen.getAllByText(subTitle).length).toBe(1);
    });
});