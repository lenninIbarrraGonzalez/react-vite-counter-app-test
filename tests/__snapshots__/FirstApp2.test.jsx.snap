// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp2 /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Este es mi titulo
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
